### DISCUSSION TITLE
Does Mixpanel session replay support canvas?

### DISCUSSION THREAD

**Discussion by sunderhus** [mixpanel/docs-legacy#1669] [Q&A] [answered]
I work on a canvas-based application. That being said, everything you see on the screen and interact with is rendered on a <canvas> tag in the web application. I tried using the current Mixpanel session replay documentation, but I couldn't get the replay to reconstruct the UI properly.

My question is: Does Mixpanel session replay support canvas? If yes, does anyone have documentation or a guide to help make it work?

If you're wondering what I see when using the current session replay:  
- I only see a blank page with the mouse moving and clicking around.  

By the way, I noticed that the underlying library Mixpanel uses to support this new functionality (rrweb) seems to support canvas-based applications through plugins:  

- [rrweb-plugin-canvas-webrtc-record](https://github.com/rrweb-io/rrweb/tree/master/packages/plugins/rrweb-plugin-canvas-webrtc-record)
- [rrweb-plugin-canvas-webrtc-replay](https://github.com/rrweb-io/rrweb/tree/master/packages/plugins/rrweb-plugin-canvas-webrtc-replay)

**Comment by themacguffinman** [comment, marked as answer]
We're adding support for canvas Soon™, it's being prepared for an upcoming JS SDK release.

Canvas elements will be snapshotted as images at up to 15 fps.

**Reply by sunderhus** [reply]
That's great to hear! Thank you for your feedback.  

Do you have an estimated release date for this feature on the JS SDK?

**Reply by themacguffinman** [reply]
We're aiming to release it by end of week, otherwise will be in the next month.

**Reply by sunderhus** [reply]
TY @themacguffinman !!

**Comment by tdumitrescu** [comment]
This is available now in v2.58.0: https://github.com/mixpanel/mixpanel-js/releases/tag/v2.58.0

**Reply by themacguffinman** [reply]
Note that you need to enable the `record_canvas` option (disabled by default) in the JS SDK: https://docs.mixpanel.com/docs/tracking-methods/sdks/javascript#init-options

**Reply by sunderhus** [reply]
@tdumitrescu and @themacguffinman I created a PR to update `@types/mixpanel-browser`, just in case someone else intends to do this too.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71470
